### ISSUE TITLE
The `button-name` markup rule throws false positive for label + for with id

### ISSUE THREAD

**Issue by emmayjiang** [dequelabs/axe-core#4625]
### Product

axe-core

### Product Version

4.10.0

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation
The following should pass the `button-name` rule:
```
<label for="my-button">Label</label>
<button id="my-button"></button>
```

The real-world use case for this is using a button with the role of a combobox.

#### Actual
The code fails the `button-name` rule.

#### How to Reproduce
https://jsfiddle.net/nvagwdhe/

#### Additional context
This is where the HTML spec defines a button as an element that can be labelled via a label: https://html.spec.whatwg.org/multipage/forms.html#category-label

**Comment by straker** [comment]
Thanks for the issue. This should now be resolved in [v4.10.1](https://github.com/dequelabs/axe-core/releases/tag/v4.10.1). Could you update to latest and see if it works?
